### PR TITLE
Wence/feature/swarm updates

### PR DESCRIPTION
**Pull request by wence-**
The vertex-only-mesh part is just the finat interface change, then I started yak-shaving.

### CODE REVIEW

**Comment by wence-** [comment]
Needs to go simultaneously with firedrakeproject/firedrake#2175.

File: tsfc/finatinterface.py [hunk 1]
```diff
@@ -117,6 +117,9 @@ def convert_finiteelement(element, **kwargs):
 
         return finat.make_quadrature_element(cell, degree, scheme), set()
     lmbda = supported_elements[element.family()]
+    if element.family() == "Real" and element.cell().cellname() in {"quadrilateral", "hexahedron"}:
+        lmbda = None
+        element = ufl.FiniteElement("DQ", element.cell(), 0)
```
Comment (ReubenHill) [inline tsfc/finatinterface.py:122]: What does this fix? Does it have any impact on https://github.com/firedrakeproject/firedrake/issues/2095 ?

**Review by ReubenHill — COMMENTED** [review]
All looks entirely reasonable. If firedrake tests pass then happy to see this merged

**Review by ReubenHill — COMMENTED** [review]
Thanks for this, all looks very reasonable. If firedrake tests pass then I'll very happily see this merged.

**Pull request merged**